http-server/src/test/java/io/airlift/http/server/TestHttpServerProvider.java

```diff
@@ -392,18 +395,33 @@ public void testShowStackTraceDisabled()
     public void testStop()
             throws Exception
     {
-        createAndStartServer();
-
+        DummyServlet servlet = new DummyServlet();
+        createAndStartServer(servlet);
         try (HttpClient client = new JettyHttpClient()) {
             URI uri = URI.create(httpServerInfo.getHttpUri().toASCIIString() + "/?sleep=50000");
             Request request = prepareGet().setUri(uri).build();
-            HttpResponseFuture<?> future = client.executeAsync(request, createStatusResponseHandler());
+            HttpResponseFuture<StatusResponse> future = client.executeAsync(request, createStatusResponseHandler());
 
+            // wait until the servlet starts processing the request
+            while (servlet.getActiveRequests() == 0) {
+                Thread.sleep(100);
+            }
+
+            // call stop when the request is still active
             server.stop();
 
+            // wait until the server is stopped
+            server.join();
```

Comment:
This can wait for a long time (close to but not exceeding the `@Test(timeOut)` value), which defeats the purpose of the test, which is to assure that the server shuts down quickly, even if there are long-running requests. We should have a timeout here.

Reply:
Thinking about this more... we actually don't know if `stop()` takes a long time. We expect it returns immediately, but it might not.

Reply:
Yep. Also we have to wait anyway until the server stops, because as you say we want to make sure that the server shuts down, without joining we can't know.